src/proof/arith_proof.cpp

```diff
@@ -1188,4 +1188,11 @@ void LFSCArithProof::printAliasingDeclarations(std::ostream& os, std::ostream& p
   // Nothing to do here at this point.
 }
 
+bool LFSCArithProof::printsAsBool(const Node& n)
+{
+  // Our boolean variables and constants print as sort Bool.
+  // All complex booleans print as formulas.
+  return n.getType().isBoolean() and (n.isVar() or n.isConst());
```

Comment:
Is this peculiar to arithmetic proofs? Or can be moved to the original function in `theory_proof.cpp`?
If we really need to override, maybe document why this is different than the general case?

Reply:
This rule is specific for arithmetic proofs. Note that `theory_proof.cpp` does contain a default implementation, but the function is virtual, because ultimately, since theories print the theory-proof, it is theories that control the type of a printed proof term.

How about I expand the documentation for `TheoryProof::printsAsBool`, elaborating on why theories may need to override?